frontend-plugin-core/src/main/java/com/github/eirslett/maven/plugins/frontend/lib/version/manager/VersionManagerLocator.java

```diff
@@ -0,0 +1,25 @@
+package com.github.eirslett.maven.plugins.frontend.lib.version.manager;
+
+import com.github.eirslett.maven.plugins.frontend.lib.InstallConfig;
+import com.github.eirslett.maven.plugins.frontend.lib.version.manager.client.VersionManagerClient;
+import com.github.eirslett.maven.plugins.frontend.lib.version.manager.client.VersionManagerFactory;
+
+import java.util.Arrays;
+
+public class VersionManagerLocator {
+
+    private final InstallConfig installConfig;
+
+    public VersionManagerLocator(InstallConfig installConfig) {
+        this.installConfig = installConfig;
+    }
+
+    public VersionManagerType findAvailable() {
+        VersionManagerFactory versionManagerFactory = new VersionManagerFactory(installConfig);
+        for (VersionManagerType versionManagerType : VersionManagerType.values()) {
+            VersionManagerClient versionManagerClient = versionManagerFactory.getClient(versionManagerType);
+            if (versionManagerClient.isInstalled()) return versionManagerType;
```

Comment:
I like trying each version manager so people can have multiple installed without issues, e.g. fnm for everything except mise for JSM. Problem is we should adjust the logs to be less noisy. Single output

Reply:
ATM it only checks if version manager directory is available but it doesn't check if node version is available in the version manager. This could be improved to choose version manager based on node versions available